### ISSUE TITLE
[RFC] Deprecate the removal of 'Bundle' suffix in Twig paths

### ISSUE THREAD

**Issue by javiereguiluz** [symfony/symfony#20130]
### Context

[This method](https://github.com/symfony/twig-bundle/blob/3.1/DependencyInjection/TwigExtension.php#L141) of the TwigBundle removes the `Bundle` suffix before adding them as Twig paths. So in your templates:

``` twig
{# it works #}
{% extends '@App/Default/index.html.twig' %}

{# it doesn't work #}
{% extends '@AppBundle/Default/index.html.twig' %}
```
### Problem
- In the particular case of the AppBundle, the `@App/...` notation may be confusing because you may not be sure if it refers to `app/...` or `src/AppBundle/...`
- This behavior is not consistent in Symfony (everywhere else you must add the `Bundle` suffix):
  - For example, to load resources in routing:
    
    ``` yaml
    # it works
    resource: "@AppBundle/Controller/"
    
    # it doesn't work
    resource: "@App/Controller/"
    ```
  - The old "bundle notation" also requires the `Bundle` prefix:
    
    ``` twig
    {# it works #}
    {{ render(controller('AppBundle:Article:recentArticles')) }}
    
    {# it doesn't work #}
    {{ render(controller('App:Article:recentArticles')) }}
    ```

In summary, the automagical removal of the `Bundle` suffix makes developers think: which is the right notation for this? Should I add the `Bundle` suffix or not?
### Proposal
- I propose to remove this feature and always use the `Bundle` suffix everywhere.
- Although you now would type 6 more characters, I think it's worth it to avoid confusion and improve consistency.
- If accepted, first I'd add both paths (with and without `Bundle`) but deprecate the Twig paths without the `Bundle` suffix. Then I'd remove the suffix-less paths in Symfony 4.0.

Thoughts?

**Comment by Koc** [comment]
relates to #11051

**Comment by iltar** [comment]
@javiereguiluz I think that's a good idea. However, your example refers to the `@AppBundle`, which you have if you follow the best-practices. Those best-practices also tell you to store your templates in `app/Resources`. So to make this clear, it's a generic bundle issue and not specific to the `AppBundle`.

**Comment by javiereguiluz** [comment]
@iltar yes ... but keep in mind that some people mix "best practices" with "own practices" and for example they store the templates in the src/AppBundle/Resources/views/ directory, so they use the `@App/...` syntax.

**Comment by fabpot** [comment]
For what it's worth, I did it on purpose to make paths shorter.

**Comment by xabbuh** [comment]
I have no prefecence whether or not to use the bundle suffix. But I am all for making it consistent. So either load resources without the suffix or add it for Twig.

**Comment by nicolas-grekas** [comment]
Another way to add consistency we just talked about on Slack with @javiereguiluz and @iltar is to always use and map the `@Foo` prefix to the FooBundle.
That should be made to work with resources (with a deprecation when `@FooBundle` is used), and also with controllers, e.g. `controller('@Foo:Article:recentArticles')`. Are there any other places where we have such shortcuts?
WDYT?

**Comment by javiereguiluz** [comment]
If Symfony is going for bundle-less apps (I don't know if this is correct!) then it may be better to always  include the `Bundle` suffix to better differentiate *"my app stuff"* and *"third-party bundles stuff"*.

In any case, the decision should be one of the following:

1. Keep the `Bundle` suffix in Twig namespaces to match the rest of syntaxes
2. Remove the `Bundle` suffix everywhere to match the syntax of Twig namespaces
3. Do nothing and keep the inconsistency (nothing is actually wrong)

**Comment by stof** [comment]
Regarding controllers, we are not using ``@.../`` currently, but ``...:``, so I don't think we need to force consistency here (removing the Bundle suffix and adding ``@`` would still not make it look the same). Changing the convention here would be a BC break for the sake of breaking BC, which is not worth the effort for the community IMO.

And regarding resources, I don't think we need to push for them to be consistent with Twig, as supported prefixes will still be different. There is nothing forcing you to use Twig namespaces only for bundles, and so you can have some Twig namespaces which don't have a corresponding prefix for resources (which are only about bundles)

**Comment by javiereguiluz** [comment]
Let's close this as "won't fix". As @stof said, this is *"a BC break for the sake of breaking BC, which is not worth the effort for the community"*.
